src/ordered_set_lang.ml

```diff
@@ -58,7 +58,15 @@ let is_standard t =
 
 module type Value = sig
   type t
-  val name  : t -> string
+  type key
+
+  module Key : sig
+    val compare : key -> key -> Ordering.t
+  end
+
+  val key : t -> key
+
+  module Map : Map.S with type key = key
 end
 
 module Make(Value : Value) = struct
```

Comment:
I haven't thought much about it but what about taking two modules, the first one `Key` for the keys (coming with a `compare` function and a `Map` module so these can be done in the original modules such as `Module.Name`) and the second one with only `t` and `key : t -> Key.t`?

Reply:
Sure, I can do it this way too. The value isn't entirely convincing to me, but let's see if it's a little neater.